src/main/java/com/atlassian/bitbucket/jenkins/internal/util/SystemPropertyUtils.java

```diff
@@ -0,0 +1,36 @@
+package com.atlassian.bitbucket.jenkins.internal.util;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class SystemPropertyUtils {
+
+    private static final Logger LOG = Logger.getLogger(SystemPropertyUtils.class.getName());
+
+    /**
+     * @param propertyName a JVM system property
+     * @param defaultValue the value to return if the propertyName is not defined or not a valid String representation
+     *                     of a long
+     * @return the parsed long value of the system property value, or the defaultValue if the system property is
+     *         undefined or invalid
+     */
+    public static long parsePositiveLongFromSystemProperty(final String propertyName, final long defaultValue) {
```

Comment:
nit: I personally have nothing against `final` on params or vars, but it's not consistent with the rest of the codebase.